### PR TITLE
feat: Add logging to loggly

### PR DESCRIPTION
**Pull request by oskarwilliams378**
Add logging to loggly

image

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -31,6 +34,15 @@ def create_app():
     oauth_client = WebApplicationClient(client_id)
     login_manager.init_app(app)
 
+    app.logger.setLevel(os.environ['LOG_LEVEL'])
+
+    loggly_token = os.environ['LOGGLY_TOKEN']
+
+    if loggly_token is not None:
```
Comment (BenRamchandani) [inline app.py:41]: `os.environ['LOGGLY_TOKEN']` will already have thrown if `LOGGLY_TOKEN` isn't in `os.environ`.

I think you want something like this, and get rid of `LOGGLY_TOKEN` in `.env.test`
```suggestion
    app.logger.setLevel(os.environ.get('LOG_LEVEL', 'DEBUG'))

    loggly_token = os.environ.get('LOGGLY_TOKEN')

    if loggly_token is not None:
```

**Review by BenRamchandani — CHANGES_REQUESTED** [review]
The logging looks good. It would be nice to log logins and user ids for actions too, but that probably covers it.

You do need to fix your travis build and the instructions for running tests in your README though and think about how you want logging in your tests to work - probably best to make `LOGGLY_TOKEN` optional and not log to loggly in your tests.

File: app.py [hunk 2]
```diff
@@ -31,6 +34,15 @@ def create_app():
     oauth_client = WebApplicationClient(client_id)
     login_manager.init_app(app)
 
+    app.logger.setLevel(os.environ.get('LOG_LEVEL', 'DEBUG'))
+
+    loggly_token = os.environ.get('LOGGLY_TOKEN')
+
+    if loggly_token is not None:
```
Comment (hugh-emerson) [inline app.py:41]: You could change this to also skip an empty string value:

```suggestion
    if loggly_token:
```

File: README.md [hunk 3]
```diff
@@ -15,10 +15,12 @@ We use docker to run this project, make sure docker is installed locally.
 
 Create copies of the `.env.template` file named as `.env`, `.env.development`, and `.env.test`.
 To these files add your mongo db connection string, and default database.
-This will setup the environment variables for both running as dev and as prod.
+This will set up the environment variables for both running as dev and as prod.
 
 The `WRITER_USER_ID` environment variable is the GitHub user id that should be the only user with Auth to be a writer.
 
+In order to log with Loggly, add the `LOGGLY_TOKEN` environment variable, which is the customer token for the Loggly logging found [here](https://oskarwilliams.loggly.com/tokens).
```
Comment (hugh-emerson) [inline README.md:22]: That link looks a little specific to you.

File: variables.tf [hunk 4]
```diff
@@ -44,6 +44,15 @@ variable "DOCKER_ENABLE_CI" {
   default = "true"
 }
 
+variable "LOG_LEVEL" {
+  description = "The log level of the app, default to ERROR"
+  default = "ERROR"
+}
+
+variable "LOGGLY_TOKEN" {
+  description = "The customer token for the loggly account"
```
Comment (hugh-emerson) [inline variables.tf:53]: Variables can have a `sensitive = true`, like outputs, to ensure they don't get included in any logs

**Review by hugh-emerson — APPROVED** [review]
You could also log a warning when your `writer_only` function finds someone unauthorised. People should only hit that if there's a bug or they are deliberately attempting to do something they shouldn't.

Otherwise I've just left some minor comments. Merge when you're ready 👍

**Pull request merged**